packages/@aws-cdk/integ-tests/lib/test-case.ts

```diff
@@ -0,0 +1,48 @@
+import * as path from 'path';
+import { IntegManifest, TestCase } from '@aws-cdk/cloud-assembly-schema';
+import { attachCustomSynthesis, ISynthesisSession, Stack, StackProps } from '@aws-cdk/core';
+import * as fs from 'fs-extra';
+import { IntegManifestWriter } from './manifest-writer';
+
+// keep this import separate from other imports to reduce chance for merge conflicts with v2-main
+// eslint-disable-next-line no-duplicate-imports, import/order
+import { Construct } from '@aws-cdk/core';
+
+export interface IntegTestCaseProps extends StackProps, TestCase {
+  testFileName: string;
+}
+
+// TODO Find a better name for this class
+class IntegTestCaseConstruct extends Construct {
+  constructor(scope: Construct, id: string, props: IntegTestCaseProps) {
+    super(scope, id);
+
+    attachCustomSynthesis(this, {
+      onSynthesize: (_session: ISynthesisSession) => {
```

Comment:
I don't think we need this logic to get the filename.

```ts
onSynthesize: (session: ISynthesisSession) => {
  const snapshotDir = session.assembly.outdir;
},

```

https://github.com/aws/aws-cdk/blob/b212ceed7f05292f94fe336aa8d8ebcff4415b8c/packages/%40aws-cdk/core/test/synthesis.test.ts#L64

Reply:
Of course. Thanks!